bam/reader.go

```diff
@@ -138,11 +136,14 @@ func (b *buffer) readUint32() uint32 {
 	return binary.LittleEndian.Uint32(b.bytes(4))
 }
 
-func newBuffer(r *bgzf.Reader, buf [4]byte) (*buffer, error) {
+func newBuffer(br *Reader, buf [4]byte) (*buffer, error) {
```

Comment:
The `buf` is needed as holds that bytes that indicate the block size. I could move that part out of Read() and into newBuffer. Actually, I think that would be nicer.